player/lua/ytdl_hook.lua

```diff
@@ -1016,6 +1020,8 @@ function run_ytdl_hook(url)
     msg.verbose("youtube-dl succeeded!")
     msg.debug('ytdl parsing took '..os.clock()-start_time..' seconds')
 
+    mp.set_property_native("user-data/ytdl/json", json)
```

Comment:
I think this should not be persistent storage and instead `script-message` to notify scripts that may be interested in it.

Reply:
The json is a bit long and it indeed doesn't make much sense to hold on to it for long. It does feel a bit odd though having to access the path through `user-data/ytdl/path` property, but access the json via an entirely different mechanism -- script-message.

Anyway, sure, I can change that. Before I do though, I would like to avoid having to change it back. Do we want to wait for someone else's input on this or am I good to make the change?

Reply:
The difference for me is that, `ytdl/path` is persistent and will not change. While `json` is an result of given extraction.

Sure you can wait for other comments to gather more opinions.

Reply:
One problem I see with storing it in `user-data` is that it should always reflect what we have for the current file. With the way this is implemented now it only gets updated when acquisition is successful, so e.g. when switching to the next playlist item, the json in `user-data` is not the correct one for the current path until we get a new one.
By also storing the path which was used for the json, scripts can check if it is for the current path.

Sending a script-message like I did in my PR avoids any race conditions where the path is updated but the json is still the old one, or the other way around.

Reply:
fwiw another option would be to store the parsed json in `user-data/ytdl/data`, so that one could e.g. use `user-data/ytdl/data/format_id` in a conditional profile.

Reply:
`file-local-options/` exists, but it has little bit different purpose.

Exporting parsed json sounds like a good idea, but since we don't have control over the format of it, it wouldn't very usable. That's why I see passing just a json and document it is directly copied from ytdl as better option. We just provide a bridge for the data to flow, not really any assurances about it.

Reply:
If the main issue with how the PR does `mp.set_property_native()` is that it doesn't always reflect what we have for the current file, then how about making it always reflect what we have for the current file? For example, have ytdl_hook delete the property on the ~~end-file event~~ `on_after_end_file` hook? That would also remove the need to communicate which url the json is for, as the property would always refer to the currently open file.

I have pushed this change right now, please check it out.

You can test with, e.g.:

```lua
local utils = require 'mp.utils'

mp.register_event("file-loaded", function()
    print("user-data/ytdl/path: " .. mp.get_property("user-data/ytdl/path"))
    print("user-data/ytdl/json: " .. utils.to_string(mp.get_property_native("user-data/ytdl/json")))
end)
```

---

Sharing the json it via `script-message` works too, but then we also need to specify the url to avoid race conditions as @christoph-heinrich has mentioned.

If we want to share the parsed json, i.e. a lua table, we can do so only via properties. Non-parsed json, i.e. a string, can be shared via both: properties and `script-message`. We could even share both parsed and non-parsed json via properties if we wanted to, though that might be a bit redundant.

So we have to decide on communicating via a property or `script-message`, and on sharing a parsed json, an unparsed json or both.